application/helpers/common_helper.php

```diff
@@ -1104,8 +1104,13 @@ function getExtendedAnswer($iSurveyID, $sFieldCode, $sValue, $sLanguage)
                             if (!isset($file['comment'])) {
                                 $file['comment'] = '';
                             }
+                            $size = "";
+                            if($file['size'] && strval(floatval($file['size'])) == strval($file['size'])) {
```

Comment:
What is the logic behind this?
` strval(floatval($file['size'])) == strval($file['size'])`

Reply:
It's not a float : it's a string currently, then : just to be sure to have a string «like» a float.

Avoit AAA , but avoid 12.1a too : it's still invalid.

Reply:
I understand. Still, disagree a bit.
On line 1110 you do `round($file['size'])`, so after all, you are expecting a float.
Why not just converting to float and then sanitizing?

This is just a feedback comment. It is not worth going back and forth.

Reply:
>  Why not just converting to float and then sanitizing?

1. Previous survey can still have invalid value. 
2. VV import can have invalid value
3. LSA import can have invalid value

New value have `strval(floatval($file['size'])) == strval($file['size'])` OK,  Except for 0 (when floatval("A") for example.